libs/vkd3d/command.c

```diff
@@ -4527,6 +4576,10 @@ static void d3d12_command_list_update_descriptors(struct d3d12_command_list *lis
     if (bindings->dirty_flags & VKD3D_PIPELINE_DIRTY_STATIC_SAMPLER_SET)
         d3d12_command_list_update_static_samplers(list, bindings, vk_bind_point, layout);
 
+    /* If we can, hoist descriptors from the descriptor heap into fake root parameters. */
+    if (bindings->dirty_flags & VKD3D_PIPELINE_DIRTY_DESCRIPTOR_TABLE_OFFSETS)
```

Comment:
I think we need a separate flag here. `VKD3D_PIPELINE_DIRTY_DESCRIPTOR_TABLE_OFFSETS` will not be set when the app calls SetDescriptorHeaps, but we want to update hoisted descriptors in that case.

Something like `VKD3D_PIPELINE_DIRTY_STATIC_DESCRIPTORS` should do, then set that when changing the CBV_SRV_UAV heap or when setting descriptor tables.

Reply:
Fixed.